sherpa/astro/io/pyfits_backend.py

```diff
@@ -263,6 +264,66 @@ def _try_vec_or_key(hdu, name, size, dtype=SherpaFloat, fix_type=False):
 
 # Read Functions
 
+# Crates supports reading gzipped files both when the '.gz' suffix
+# is given and even when it is not (i.e. if you ask to load 'foo.fits'
+# and 'foo.fits.gz' exists, then it will use this). This requires some
+# effort to emulate with the astropy backend.
+
+def is_binary_file(fname):
+    """Do we think the file contains binary data?
+
+    Parameters
+    ----------
+    fname : str
+       The file name.
+
+    Returns
+    -------
+    flag : bool
+       ``True`` if the file appears to contain binary data,
+       ``False`` otherwise.
+
+    Notes
+    -----
+    This is a wrapper around the version in sherpa.utils which
+    attempts to transparently support reading from a `.gz`
+    version if the input file name does not exist.
+    """
+
+    if not os.path.exists(fname):
+        fname = fname + '.gz'
```

Comment:
Note that if fname does not exist and fname+gz does not exist either, is_binary_file will tell you that fname+gz does not exist, which is confusing, because you never asked to load fname+gz. I have a patch for this, I am just trying to figure out whether it makes sense to add a unit test, given that we still can't use mock (it's in the process of being approved as a CIAO OTS).